### ISSUE TITLE
Social Login not collecting user info

### ISSUE THREAD

**Issue by sagivf** [driftyco/ionic-cloud#20]
Most auth systems collect the social providers user data (name, email at times, etc...).
At least for google it doesn't seem to do so.

**Comment by dwieeb** [comment]
@sagivf -- @devnetics is actually working on this right now. None of the current social providers are collecting additional information. This was a known limitation that we are now addressing. I'll keep this open to inform you when you can expect this functionality.

**Comment by IndyJones72** [comment]
@dwieeb  - Ok so even though this was for Google, we'll at the very least hopefully have this information for Google, Facebook and Twitter?

**Comment by dwieeb** [comment]
@IndyJones72 Yes! I changed the title so it is more fitting.

**Comment by sagivf** [comment]
@dwieeb I've been wondering - will the platform do some sort of unification of the same person with different social logins or will each one create a new user in the system?

Not sure myself which is better, I'm guessing they could be unified based on the email.  Probably best if it were configurable.

**Comment by dwieeb** [comment]
@sagivf Unification of users is actually pretty complex. Even once you're sure they're the same person (via email, for example), merging data and avoiding data destruction is a complicated process. Not all social networks give us email addresses and most don't give enough general data to piece things together. 

But honestly, I don't see _logins_ being merged. If a user shares an email address in two social networks and logs in to your app with one network then the other, they would expect to have a completely new account, not their data from the other. Best to keep them as separate accounts.

That being said, data analysis on users can be done while maintaining the two separate accounts. This is something we're interested in looking at later. But, obviously there's much to do before that.

**Comment by sagivf** [comment]
@dwieeb I agree with all of what you say.
Take into consideration that there are use cases were you want a user to sign in one way. then validate his email, then add social logins for all sorts of info from the social provider.

This is defiantly not a common use case for apps and not necessary to do with authentication, but it's good to keep in mind.

**Comment by dwieeb** [comment]
@sagivf Ah, yeah. Not a use case we'd tackle with our authentication. Devs are free to add those additional hookups themselves and then save such information in users' custom data. I would say it's not in our scope to add support for that, something which could be done with plugins such as [cordova-plugin-facebook4](https://github.com/jeduan/cordova-plugin-facebook4) or libraries.

**Comment by jaspervanveghel** [comment]
Are there any updates to report on the approximate completion date of this yet? We would like to integrate this into our app, which we'd like to release at the end of this month.

**Comment by devnetics** [comment]
@jaspervanveghel This will be released next week. I'll update here when it's live.

**Comment by IndyJones72** [comment]
@devnetics - great news. We are also looking to release at the end of the month and this will be great to have

**Comment by ArkangelB** [comment]
@devnetics did you have any news :) we also have a release  but next month and give full facebook login with complete user data will be great!

**Comment by devnetics** [comment]
Hey guys, we have the PR ready to go on the backend to support Oauth provider user data. We didn't want to turn on a fairly complex new feature Friday afternoon, so we're hitting the on switch Monday morning. I'll post here when it's live. 

We are giving you access to the basic user details (such as username, email (when available), full_name, and profile_picture), and then also recorded a raw_data object that gives you access to any of the other data that that particular provider happens to return on the Users/Self endpoint (for example, Github returns things like `location`, `created_at`, `hireable`, etc.). 

Note: We are not providing access to the provider tokens, that is still under discussion.

**Comment by cfjedimaster** [comment]
Can I ask why you don't want to (possibly) provide the token?

**Comment by IndyJones72** [comment]
@devnetics - Great news, and I completely understand about throwing the switch on a Friday afternoon.  Really looking forward to this on Monday.

**Comment by sagivf** [comment]
@devnetics Thanks for working on this.

I have recently implemented something similar for a wen app and was wondering regarding keep the user data up to date - meaning what happens if a user changes his photo / email / name etc... It's not critical for most apps, but could be in some cases.

Is there any implementation of this?
If not perhaps suppling the token will at least give the developers the option.

A few words of advice:
1) Make sure that after a user logs back in you override the data with the new results.
2) Obviously you can update the user data in a background Job, but what I do is only update once an hours for active users (ones using the api).
2) Twitter - you need a OAuth for any data. You can use a general app token but there is a rate limit. The good news is that twitter users tokens don't expire.
3) Facebook - you can get most data with a simple GET request with an access_token. Some fields are available with a simple app access_token, but some (like email) you need the users access_token and that has an expiry (which can be extended with a refresh token).
4) Facebook Image - that can easily be fetched dynamically by the userId.
 

Let me know if you I can help in any way and thanks again :)

**Comment by ArkangelB** [comment]
Hi @devnetics, does the PR it's ready to update or it will take a little more to be available?

**Comment by devnetics** [comment]
@ArkangelB We soft launched it yesterday. The new docs and cloud client are also on their way, which will give you local access to the provider data (on log in with the cloud client), and the docs will demonstrate how to use it. If you're feeling antsy and want it right now, you can use the latest version of the cloud client [github.com/driftyco/ionic-cloud](github.com/driftyco/ionic-cloud), but I would recommend waiting for our next publish to npm for any production level applications. 

@sagivf Thanks! We already have it updating user's. Any data that is changed gets overridden, anything that did not change (say for example the new set of data that comes in doesn't have email, but the old set does have email) stays the same. Currently this happens on user login.

**Comment by IndyJones72** [comment]
@devnetics - just a quick shoutout...this is working pretty darn well.  

I share a few of our fellow posters in that having the token would be useful so we could obtain the email address of the user, however...and I'm reaching here...is it possible to send an email to Facebook somehow incorporating the user's FB id?  If so, maybe we don't need it and not have to worry about the token and added layers of difficulty that FB may add.  

Just wanted to say thank you for getting this adjusted and out there.  We should be almost set to release at the end of the month with this.

**Comment by sagivf** [comment]
Thanks @devnetics, great to see this coming along.
But do you pull to update the user details (like changing his name or photo)? Or does the user have to re login?

**Comment by agarwalswapnil** [comment]
@devnetics please suggest when will it be available for public npm ?
I am on ionic version 2.0.0

**Comment by IndyJones72** [comment]
Ok am I reading the code incorrectly, or shouldn't the new social login info be available in $ionicUser.social after a successful login?

**Comment by sagivf** [comment]
@IndyJones72 I may be wrong here but I think this is only for Ionic2, although this feature could probably easily be back ported as it's mostly server side.

**Comment by IndyJones72** [comment]
@sagivf - oh man I feel dumb then. I had thought ionic cloud was for both platforms as I did the whole migration from web platform to ionic.cloud and everything seemed to work so that's why I had asked. 

I do see the "data" in my network debug feed and see everything coming back from the social service. However I can't use that data for some reason in the promise...all I get is the uuid. So I'm sure I'm doing something wrong here. 

How difficult is it to convert your ionic 1 to ionic 2?  I know there's documentation out there so not looking for details. Just curious as to the effort involved.

**Comment by sagivf** [comment]
Actually you maybe right @IndyJones72, I guess your using the compiled results of typescript source... 

I re-wrote my app since I had a lot of business changes due anyway.

My advice would be:

1) Ionic2 and angular2 are very different then there predecesores so conversion would probably be very tricky.

2) Writing the code Is usually much simpler then then getting your UX flow and business logic in place. So re-rights are not that big of a deal unless you have a very sensitive, monologic business with little documentation and code which no be knows what it does.

3) Ionic2 is much faster to write. They've locked down a lot of good concepts and the TypeScript intelisence is very helpful.

4) If your using Ionic-cloud that takes away a lot of complications like login/push/etc...

Goo luck either way :)

**Comment by agarwalswapnil** [comment]
@sagivf I believe data for social authentication will be available for ionic 1.3.
The documentation http://docs.ionic.io/docs/user-usage is available for ionic 1.3.

**Comment by dwieeb** [comment]
Here's how this works. The Cloud Client initiates a series of HTTP redirects through our servers and the social network. At this time we store the social data, such as name, email, and profile picture on the user. The user is returned to the Cloud Client and that data will be available under `$ionicUser.social`, as @IndyJones72 mentioned. Our new documentation will have more details and you can wait for it if you're struggling.

@sagivf Ionic Cloud Client doesn't rely on the Ionic Framework. You can use it in Ionic 2 _and_ Ionic 1 apps. The _integration_ of the client into those frameworks may differ, but our new documentation will feature both.

@sagivf @IndyJones72 Perhaps it's best to wait for another npm release. I'm planning on one today or tomorrow.

@sagivf That's very good advice. 👍 I agree with all of it.

@agarwalswapnil Those are our current docs, which won't mention anything about this new feature.

**Comment by IndyJones72** [comment]
@dwieeb - Thanks so much for the reply and the clarification.  If you have another npm scheduled today, I probably will wait for that...install it since it seems the compiled version I have is missing the social part of that new user object.  Please let us know when it's out there so I can give it a whirl.  Thanks!

**Comment by ArkangelB** [comment]
@dwieeb  Thank you i'll be wait it for your npm release!

**Comment by dwieeb** [comment]
@sagivf @IndyJones72 @ArkangelB I just published to npm! Please test social auth user data. Some things have changed--deploy in particular. Please see [this diff of the migration guide](https://github.com/driftyco/ionic-cloud/wiki/Migration-Guide/_compare/e23dee25d06bb56bfefb9aefe3095397269884ce...5769cac73176c71a9f37ba6b497c7084f96bd298) for more details.

**Comment by IndyJones72** [comment]
@dwieeb - On it!  Will report back.

**Comment by ArkangelB** [comment]
@dwieeb awesome! will be in touch!

**Comment by IndyJones72** [comment]
@dwieeb @sagivf @ArkangelB - Installed latest npm, however can't get to the login page of twitter or Facebook.  Anyone else having issues?  I didn't change anything and my team didn't commit any new changes, so it should be the way it was always working.  Did I miss a change to the login?  Here's my current way I call the login....

`    $scope.FacebookLogin = function () {
        $ionicAuth.login('facebook', { 'remember': true }).then(function (_info) {
            $scope.loginSuccess();
        }, function (_info) {
            $scope.signupFailure(_info);
        });
    };`

Am I missing something that has changed?

**Comment by dwieeb** [comment]
@IndyJones72 Looks correct (but you should remove `{'remember': true}` as it has no effect). 

What are you seeing? It should resolve or reject that promise. Any errors?

**Comment by IndyJones72** [comment]
@dwieeb - Ok removed that and still not reaching it.  I'm going to reboot....see if that makes any difference and then see if I can't capture any errors via the console or Safari and report back.  Probably something on my side, but did want to reach out and see if anyone else was having any issues before heading down that all-too familiar rabbit hole of my own stuff breaking it.

**Comment by dwieeb** [comment]
Did you do `cordova plugin add cordova-plugin-inappbrowser --save`?

**Comment by ArkangelB** [comment]
can somebody help me i'm trying to do the update but my npm update fails someone have the same issue? or can somebody tell me what am i doing wrong updating?

**Comment by dwieeb** [comment]
@ArkangelB What is the command you're running and the error message?

**Comment by fknop** [comment]
@dwieeb Hi, thanks for this feature, working for me with Google+.

**Comment by dwieeb** [comment]
@ArkangelB I'm not sure what all that's about. Try updating just our package. `npm update @ionic/cloud`

@fknop Awesome, glad to hear. 😄

**Comment by ArkangelB** [comment]
@dwieeb haha sorry man i really was too busy and don't even look the  entire document that you make for us, an apology i will initiate the update 👍  

Thanks for everything!

**Comment by ArkangelB** [comment]
Hello, It exist a typo error at the tutorial,  you change  **ionic.cloud.min.js** for **ionic-cloud.min.js** so it does not load the module because not exist that file..

**Comment by dwieeb** [comment]
@ArkangelB Thanks, fixed that in the migration guide!

**Comment by ArkangelB** [comment]
hey @dwieeb i was trying to get Social Profile information from Facebook i can't find the data.. this is what am i using: 

`$scope.facebookLogin = function() {           
        $ionicAuth.login('facebook', authSettings).then(success, failure);             
    };`

Everything works as usual but i have no data to display for social provider, when i use  **ionic run ios -l** and watch my console info showing **$ionicUser**. (Safari console)

![from safari ios debug](https://cloud.githubusercontent.com/assets/20997658/17796207/dcdf323c-6573-11e6-9a8c-7428d953eb78.png)


If i do a basic login it shows user info custom you can see differents kinds of objects between social and basic info. (Chrome console).

![from ionic serve chrome](https://cloud.githubusercontent.com/assets/20997658/17796210/e26d8b68-6573-11e6-880a-11a5051ea936.png)


Does anybody has been able to get data from Facebook? you were trying with  facebook right @IndyJones72 have an idea why is not working with me?

**Comment by devnetics** [comment]
@ArkangelB Can you fully expand that `social` object and post here with what it contains?

**Comment by IndyJones72** [comment]
@ArkangelB - Correct, I had Facebook and Twitter both "showing" the data coming back in the version just before this latest NPM.  I just didn't have the new version where that would get placed automatically into the `$ionicUser.social`.  

However, for some reason since then I haven't been able to get Facebook or Twitter to open in the iOS Simulator or on an actual iPhone 6 Plus.  

I'm going to pop Google Plus on there and see if that works, if not, I made a backup just prior to the npm update so I'll go back into that folder, and retrace steps to see what the hell happened.

**Comment by IndyJones72** [comment]
@dwieeb @devnetics - Ok I found what the problem was.  Coming from the ionic-platform-web-client and following the migration guide, I didn't uncomment this line:

`<script src="cordova.js"></script>`

...which I know is stated in the index.html file, but maybe it can be put as a reminder into the migration docs for twits like me?

So @dwieeb / @devnetics - when I close the app and reload it...it knows I was already authorized by FB and bypasses my login area.  However, going to the account page where my social data is, the `$ionicUser.social` is empty.  So.....what's the best way for making sure that data is populated so I can show their avatar/image, username, etc?

**Comment by dwieeb** [comment]
@IndyJones72 Sorry for the confusion. The new deploy plugin does require the `cordova.js` include to be there. 

@IndyJones72 @devnetics I will look into this tomorrow. The social data should be persisted into local storage. If it isn't, no problem, I'll add that logic. It will be simple. Thanks for your patience during this beta period and migration process.

**Comment by IndyJones72** [comment]
@dwieeb - No worries.  I made a decision for my company and my customers over a year ago after doing over a year of research of various multi-platform frameworks  to commit to Ionic as the only one I allow my devs to use.  So we see your commitment and communication as a sure sign that I made the correct decision.

Keep up the great work and thanks for looking into that persistence.

**Comment by dwieeb** [comment]
@IndyJones72 Thank you! That means a lot. We are totally committed to making Ionic Cloud awesome. I think this cloud client and our new documentation will continue to show that.

**Comment by fknop** [comment]
@dwieeb Just one thing tho. Is there any reason why the email is null with Google+? I only have the full name and picture.

**Comment by IndyJones72** [comment]
@fknop -  I noticed the same on the FB one as well. Perhaps they're linked and hopefully while investigating the persistence issue that will be looked at as well.

**Comment by ArkangelB** [comment]
@devnetics here is both objects expanded

from iOS debug with safari console: (ionic Social Provider)

![object from safari console](https://cloud.githubusercontent.com/assets/20997658/17816662/38d1d092-65ef-11e6-9bb7-1a0d23062fee.png)

from Chrome console: (ionic basic)

![object from chrom console](https://cloud.githubusercontent.com/assets/20997658/17816686/5b932036-65ef-11e6-82a7-43d39a4d8361.png)

This is my index.html:
Maybe i have some error in there.

![index info](https://cloud.githubusercontent.com/assets/20997658/17816745/a06704fc-65ef-11e6-9ac5-35013dc7de04.png)

**Comment by devnetics** [comment]
@fknop @IndyJones72 @ArkangelB With google it seems to be coming through on `raw_data` but isn't properly being set in the local `social.data.email` property, whereas with Facebook we don't seem to be getting it at all.

Looking into this now, will update when we have the solution out.

**Comment by IndyJones72** [comment]
@devnetics - ok cool.  Any thoughts/updates on the persistence of the social data in the $ionicUser object?

**Comment by sagivf** [comment]
@IndyJones72 @dwieeb @devnetics  sorry for adding confusion regarding Ionic2, that's what https://github.com/driftyco/ionic-cloud-angular is for. After not touching this for a while I got a little confused...

I've tested out facebook, google and twitter.
1) I also think that email, name, image should be perssited to the user object.
2) Google and twitter don't supply the email, 
For facebook I think it needs to be a parameter added the the facebook request.
For twitter you need to request it on the apps settings -
![image](https://cloud.githubusercontent.com/assets/1857466/17832246/b1c695ba-6706-11e6-96ff-ebb3f4fa132f.png)
3) Any chance we can see this data on the https://apps.ionic.io and via api? I t would make it easier to test and we will probably need it there anyway.

**Comment by devnetics** [comment]
@sagivf we will be adding this to the dashboard fairly soon. The Google email fix will go out on Monday.

**Comment by devnetics** [comment]
@sagivf Twitter is a bit more complicated because you have to be approved at an app level not an API level so we can't actually provide that. You would have to get your individual app approved by Twitter. If you do, the email should be available within the social date object as normal.

**Comment by ArkangelB** [comment]
@devnetics what about facebook, did you have something about it?

**Comment by IndyJones72** [comment]
@dwieeb @devnetics - We may have already touched on this earlier, but it would be really great if the "typical" user data is persisted not only in the $ionicUser.social object, but also in the typical $ionicUser.details.image, $ionicUser.username, etc.  

Right now (unless I'm doing something wrong), I have to check to see if the $ionicUser.social is populated....and then go further into seeing if it's Twitter, or Facebook data, and then make a decision on my view as to where to pull the user's data from.

I hope that makes sense, but then again it is getting late here.  ;)

**Comment by devnetics** [comment]
@ArkangelB Facebook should be loading up that email address now. If it isn't, let me know.

@IndyJones72 The `social` object design won't be changing, as it gives us a lot of flexibility and simplicity for adding new features (such as **maybe** multiple providers). However, I think we could load those base details for you as well. @dwieeb and I will discuss and update you here.

**Comment by devnetics** [comment]
Full notes to all for what to expect this week:

1. Facebook should be getting you an email address now. There should be a secondary permissions screen (during the inappbrowser dialogue) that asks permission for the user's email address. 
2. We'll be pushing out an update that fixes several fields that appear to be null after login (Google email field, Facebook profile_picture field, and 1-2 others). 
3. While not entirely related to this issue, please note that we are working on a fix that will not require your users to re-login all the time with Facebook. See https://github.com/driftyco/ionic-cloud/issues/23

And thank you all for the continued feedback, it's been a huge help.

**Comment by IndyJones72** [comment]
@devnetics, no worries.  Right now I'm simply taking that info after login and placing it into those areas of ionicUser so I have some consistency across my logic.

However, a larger question is....while I know some things may change as stated in the guides, I am planning on releasing this app in the next week here.  How ok are we with doing this?

**Comment by ArkangelB** [comment]
@devnetics will be working with this right now, it just a **npm update @ionic/cloud** or i will need to make part of the migration that you send?

**Comment by edjeezy** [comment]
Hello guys i am a roockie on ionic cloud and i would like to get a code snippet on how to get user's name and picture from facebook . Also a link to the "new" docs would be really helpfull .Thanks a lot !

**Comment by sagivf** [comment]
@edjeezy have you implemented the Facebook login? 
If you can wait, I would wait for the docs to be released and the api locked down.
But currently you should be able to find the details under the `user` object in the `social` property.

Also you can get the image based on the user id.`https://graph.facebook.com/v2.7/${user.id}/picture`

**Comment by edjeezy** [comment]
@sagivf can you give me a sample code

**Comment by sagivf** [comment]
@edjeezy Are you using Ionic2 or Ionic1?

**Comment by edjeezy** [comment]
ionic 1 but i am going to upgrade to 2  for the production version

**Comment by sagivf** [comment]
So code sample For which :) ?

**Comment by edjeezy** [comment]
v1

**Comment by sagivf** [comment]
I'm using V2 - but I think this should basically work - 

```
Ionic.Auth.login('facebook').then(function(user){
  var name = user.details.name;
  var picture = user.social.facebook.data.raw_data.picture.data.url; // this will defiantly change
});
```
http://docs.ionic.io/docs/user-usage

But again this is what this ticket is about and will probably change soon so keep following.

**Comment by IndyJones72** [comment]
@edjeezy - I'm using v1 and this works for my app so far.  Just waiting on some sample code for implementing the Push Notifications and Deploy/Check for Updates and I can finally package this thing up for Apple Police review.  Let me know if you have any questions.

`$scope.FacebookLogin = function () {
        $ionicAuth.login('facebook').then(function (_info) {

            $ionicUser.details.username = $ionicUser.social.facebook.data.full_name;
            $ionicUser.details.email = $ionicUser.social.facebook.data.email;
            $ionicUser.details.image = $ionicUser.social.facebook.data.raw_data.picture.data.url;
            $ionicUser.save();

            $scope.loginSuccess();
        }, function () {
            $scope.signupFailure(_info);
        });
    };`

**Comment by edjeezy** [comment]
Thanks for the support  @sagivf . One last question @IndyJones72 can i replace this code
`$ionicUser.details.username = $ionicUser.social.facebook.data.full_name;`
with this one 
`$scope.user.username=$ionicUser.social.facebook.data.full_name;` for the two way data binding or should i use another method

**Comment by IndyJones72** [comment]
@edjeezy - I don't see why not.  The only reason I do it this way is because (correct me if I'm wrong @devnetics / @dwieeb) is $ionicUser is now "global" throughout the application.  

So inside of one of my controllers, I do the following (if someone has a better way please chime in): 

`$scope.user = $ionicUser` 

Which will now allow you inside of your template to do something such as:

```
<div class="item item-avatar">
                    <img ng-src="{{user.details.image}}">
                    <h2 style="color:#333;">{{user.details.username}}</h2>
                    <p>{{user.details.email}}</p>
                    <h5>UID: {{user.id}}</h5>
                </div>
```

Now again, the reason for this is I have about 14 controllers in this app, but hopefully this makes sense.

**Comment by ArkangelB** [comment]
Hello @dwieeb  i was working to make ir work the facebook info but i looks like the problem its not ionic cloud, i update to ionic v2 and when i make a deploy **ionic run ios** the app does not update at all, so i think that its why i cant see the social object.

So did you have an idea why does this happens or how can i fix it? i was looking around but there is not so much information about ionic v2, do i have to make a downgrade of my ionic to fix the problema?

Thanks, i'll keep looking around for an answer.

**Comment by sagivf** [comment]
@ArkangelB Just so you know, I've always had problems with `ionic run ios` usually doesn't work for me. I use the emulator or run it through xcode (xcode doesn't always run smoothly either).

**Comment by sagivf** [comment]
There are so many variables to this - Ionic cli version, cordova version, node version, ionic version, ios-deploy, ios-sim, os version, plugins, xcode version and probably more :)

**Comment by ArkangelB** [comment]
@sagivf  haha it's the first time that i have problems with ionic run ios but i know what you mean, i just make a package just to know if facebook social object its working and it works so i'm happy 👍  now i can keep working  with the next stage..

**Comment by edjeezy** [comment]
Can i know when the 'updated' docs will be available ?

**Comment by devnetics** [comment]
Facebook and Google fixes are live now, the null fields should be corrected. We also pushed out a fix where in some cases Google email wasn't getting fetched properly. 

@edjeezy We're working as fast as we can to get the updated docs out. The wait shouldn't be too much longer.

**Comment by edjeezy** [comment]
@devnetics thank you for the hard work . Ionic is really a dream came thrue so keep on going .

**Comment by devnetics** [comment]
@edjeezy Thanks! It means a lot to us to hear that.

**Comment by ArkangelB** [comment]
@devnetics i have a question  i can have the image from facebook like this 

$scope.imgProfile ="http://graph.facebook.com/v2.7/10153521906500989/picture"

and it get the same photo quality  that the picture that you bring to the object but if you ask for this one 

$scope.imgProfile ="http://graph.facebook.com/v2.7/10153521906500989/picture?type=large"

it gets a much better quality, so my question is, it exist a way that we can ask for a better photo quality?

**Comment by devnetics** [comment]
@ArkangelB We provide you with the base url, so you would just need to append that `type=SIZE` param onto your url. 

For a very simple example, you could do:
`var profile_pic = $ionicUser.social.facebook.data.profile_picture + '?type=large'`;

**Comment by ArkangelB** [comment]
@devnetics  ok this still works pretty nice!

Thanks for everything!

**Comment by devnetics** [comment]
Hey guys, we are going to ahead and close this issue. We are still making small tweaks/updates, nothing breaking. We have taken into account all of the feedback and will be iterating on this feature further after release of the new docs. Please report any actual errors in a new issue.

**Comment by brudrigo** [comment]
sorry, but how can I use it in ionic2? I'm using ionic-cloud-angular but there isn't User.social.facebook.data for example. I try update it with `npm install --save @ionic/cloud-angular` but still...

**Comment by cfjedimaster** [comment]
Just checking in - was this addressed?

**Comment by IndyJones72** [comment]
@cfjedimaster - Yes it sure was.  We were able to get the basic profile information needed, but I believe there's some additional information brought back which could be helpful as well now.
